core/src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTests.java

```diff
@@ -270,7 +330,7 @@ public void testDateRangeBoundaries() throws IOException {
                 "        }\n" +
                 "    }\n" +
                 "}\n";
-        Query parsedQuery = parseQuery(query).toQuery(createShardContext()).rewrite(null);
```

Comment:
well today if you rewrite you get what used to be `rewrite(null).rewrite(null)` since we had an indirection. it's gone so if you really rewrite and you have a termrange you get a MTQ which is not expected